packages/flutter_tools/lib/src/emulator.dart

```diff
@@ -57,6 +60,136 @@ class EmulatorManager {
     return emulators;
   }
 
+  /// Return the list of all available emulators.
+  Future<CreateEmulatorResult> createEmulator(String name) async {
+    if (name == null || name == '') {
+      const String autoName = 'flutter_emulator';
+      // Don't use getEmulatorsMatching here, as it will only return one
+      // if there's an exact match and we need all those with this prefix
+      // so we can keep adding suffixes until we miss.
+      final List<Emulator> all = await getAllAvailableEmulators();
+      final Set<String> takenNames = all
+          .map((Emulator e) => e.id)
+          .where((String id) => id.startsWith(autoName))
+          .toSet();
+      int suffix = 1;
+      name = autoName;
+      while (takenNames.contains(name)) {
+        name = '${autoName}_${++suffix}';
+      }
+    }
+    
+    final String device =  await _getPreferredAvailableDevice();
+    if (device == null)
+      return new CreateEmulatorResult(
```

Comment:
nit: the formatting here looks off. 2 space indent? no trailing comma?

Since this is a newer file - added by you recently - you may just want to dartfmt it as a level-set.

Reply:
Done (though I tweaked some of its changes where they seemed to make it less readable).